docs/rfcs/20220208_admin_api_basic_auth.md

```diff
@@ -0,0 +1,234 @@
+- Feature Name: HTTP Basic Auth for admin API
+- Status: in-progress
+- Start Date: 2022-02-08
+- Authors: John Spray
+- Issue: https://github.com/redpanda-data/redpanda/issues/3702
+
+# Executive Summary
+
+Enable admin API clients to authenticate using SASL username/password credientials
+passed into an HTTP "Basic" Authentication header. 
+
+These checks will be off by default for backward compatibility, but enabled during
+setup of new clusters.
+
+## What is being proposed
+
+A cluster configuration property:
+* `admin_api_require_sasl` (boolean, default false).
```

Comment:
Will authentication be SASL based or is that going to be Basic Auth ? If basic auth i would not name it sasl

Reply:
The naming is kind of awkward, because i want to communicate that we're talking about the SASL usernames rather than just some other type of username, but as you say it's not actually SASL.

I considered "require_username" or something like that, not sure what the neatest alternative is

Reply:
Ended up changing this to `admin_api_require_auth`